lib/request_headers_middleware/delayed_job.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'delayed_job'
+
+module Delayed
+  class PerformableMethod
+    attr_accessor :object, :method_name, :args, :store
+
+    def encode_with(coder)
+      coder.map = {
+        'object' => object,
+        'method_name' => method_name,
+        'args' => args,
+        'store' => store
+      }
+    end
+  end
+end
+
+class DelayedRequestHeadersPlugin < Delayed::Plugin
+  callbacks do |lifecycle|
+    lifecycle.before(:enqueue) do |job|
+      obj = job.payload_object.dup
+      obj.instance_variable_set(:@store, RequestHeadersMiddleware.store)
+      job.payload_object = obj
+    end
+
+    lifecycle.before(:perform) do |worker, job|
+      RequestHeadersMiddleware.store = job.payload_object.instance_variable_get(:@store)
+
+      tag_logger Delayed::Worker.logger
+      tag_logger ::Rails.logger unless dj_use_rails_logger
+    end
+
+    lifecycle.after(:perform) do |worker, job|
+      untag_logger Delayed::Worker.logger
+      untag_logger ::Rails.logger unless dj_use_rails_logger
+
+      RequestHeadersMiddleware.store = {}
+    end
+  end
+
+  def self.dj_use_rails_logger
+    ::Rails.logger == Delayed::Worker.logger
+  end
+
+  def self.tag_logger(logger)
+    uuid = RequestHeadersMiddleware.store[:"X-Request-Id"]
+    logger.push_tags(uuid) unless uuid.nil?
```

Comment:
This is dangerous because it adds the dependency to TaggedLogging that is only there for Rails or better ActiveSupport. Plus what happens if we do JSon Logging. I think also this needs to be moved into some pluggable env. I tried to solve it in my message_queue PR.